src/pages/EntityDetails/Model/Secrets/SecretsTable/SecretsTable.tsx

```diff
@@ -0,0 +1,102 @@
+import { ModularTable } from "@canonical/react-components";
+import { useMemo } from "react";
+import { useParams } from "react-router-dom";
+import type { Column } from "react-table";
+
+import LoadingSpinner from "components/LoadingSpinner/LoadingSpinner";
+import RelativeDate from "components/RelativeDate";
+import type { EntityDetailsRoute } from "components/Routes/Routes";
+import {
+  getSecretsLoading,
+  getSecretsLoaded,
+  getModelSecrets,
+  getModelUUIDFromList,
+} from "store/juju/selectors";
+import { useAppSelector } from "store/store";
+
+export enum TestId {
+  SECRETS_TABLE = "secrets-table",
+}
+
+const COLUMN_DATA: Column[] = [
+  {
+    Header: "Name",
+    accessor: "name",
+  },
+  {
+    Header: "ID",
+    accessor: "id",
+  },
+  {
+    Header: "Revision",
+    accessor: "revision",
+  },
+  {
+    Header: "Description",
+    accessor: "description",
+  },
+  {
+    Header: "Owner",
+    accessor: "owner",
+  },
+  {
+    Header: "Created",
+    accessor: "created",
+  },
+  {
+    Header: "Updated",
+    accessor: "updated",
+  },
+  {
+    Header: "Actions",
+    accessor: "actions",
+  },
+];
+
+const SecretsTable = () => {
+  const { userName, modelName } = useParams<EntityDetailsRoute>();
+  const modelUUID = useAppSelector(getModelUUIDFromList(modelName, userName));
+  const secrets = useAppSelector((state) => getModelSecrets(state, modelUUID));
+  const secretsLoaded = useAppSelector((state) =>
+    getSecretsLoaded(state, modelUUID),
+  );
+  const secretsLoading = useAppSelector((state) =>
+    getSecretsLoading(state, modelUUID),
+  );
+
+  const tableData = useMemo(() => {
+    if (!secrets) {
+      return [];
+    }
+    return secrets.map((secret) => ({
+      name: secret.label,
+      id: secret.uri.replace(/^secret:/, ""),
```

Comment:
Was wondering if maybe it might make it more user-friendly if we use a `Tooltip` here in order to show the full id? Maybe we could also do something similar with "owner" as well, considering that it takes a lot of vertical space in the table?

Reply:
These are good suggestions. The ID field is going to have a copy button to get the full URI ([WD-8396](https://warthogs.atlassian.net/browse/WD-8396)) and I thought I'd leave other responsive improvements until after all the data is in the table so that I can see what needs adjusting ([WD-8521](https://warthogs.atlassian.net/browse/WD-8521)).